assets/terraform-modules/aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl

```diff
@@ -82,6 +82,8 @@ systemd:
           -v /opt/cni/bin:/opt/cni/bin:ro \
           -v /usr/lib/os-release:/etc/os-release:ro \
           -v /usr/sbin/iscsiadm:/usr/sbin/iscsiadm:rw \
+          -v /lib64/libisns.so.0:/usr/lib/x86_64-linux-gnu/libisns.so.0 \
```

Comment:
Are those libraries present in kubelet image we use? I wonder if this override may affect some other binaries in the image.

Reply:
And anyway this will break next week (earlier or later) when the dependencies change. Please use one approach I suggested in https://github.com/kinvolk/Flatcar/issues/293#issuecomment-742671025 (I think the wrapper script with `nsenter` is quite feasible).

Reply:
@pothos the problem with using nsenter trick is that we don't call iscsiadm explicitly, kubelet does that. Would you suggest running kubelet with nsenter?

I think using the `ldd` trick is feasible.

Reply:
@surajssd I think it should be enough to create a file called `iscsiadm` in `kubelet` container which is a script instead of a host binary. Probably we should write it using Ignition and then bind-mount it.

I wouldn't run entire `kubelet` with `nsenter`.

Reply:
Yes, the kubelet can't be run with nsenter because the kubelet file does not exist on the host. It's about a wrapper script called `iscsiadm` which would call it.

Reply:
@invidian @pothos could you give it a try now? What do you think of the current approach.

The kubelet systemd process's `ExecStartPre` copies the libs to a certain dir which is mounted inside the Kubelet pod and this Kubelet pod copies those files before starting the Kubelet process.

Reply:
@surajssd what is wrong with the wrapper script approach? Current approach seems like a hammer...

Reply:
It worked, created a new image. I guess I was trying to fix using something that didn't need creating a new image. But anyway, I have created an image with the new script.

Reply:
if we would have it here, it would be better documented. Do we have a repository for image building?

Reply:
Do you mean I should add something more apart from the README?

Reply:
Ah, I didn't see the README. But why can't we deploy a script using Ignition?

BTW, for kubelet images, it would be nice if we could build a multi-arch manifests instead of having separate tags.

Reply:
How would ignition put the file inside kubelet pod. Yes we can do multiple volume mounts. But doesn't this provide us more control over image now and smaller ignition which has been a reason for failure on AWS?

Added multi arch image building instructions.